Makefile

```diff
@@ -106,8 +106,12 @@ deploy:
 	$(PGO_KUBE_CLIENT) apply -k ./config/default
 
 # Deploy the PostgreSQL Operator locally
+# Note: using `--server-side=true --force-conflicts` when applying the K8s objects in order to remove the
+# `kubectl.kubernetes.io/last-applied-configuration` from the CRD since it was violating the limit
+# on size of `metadata.annotations`
+# - https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go#L36
```

Comment:
I wouldn't have thought of this, especially since the ticket specifies to add a new field in dataSource, but since `postgresCluster` and `pgBackRest` are so similar, we could... combine them with some enum flag that tells the operator whether we're pulling from a local & existing cluster (in which case, copy over the secrets and config) or from a remote cloud-based cluster (in which case, create a new config from the global and mount the given secret).

But unless we're going to say "we can't add new fields to the spec", I see us running into this limit some other day.